File: app/src/main/java/com/example/newstart/BaseActivity.java
package com.example.newstart;

import android.content.Context;
import android.os.Bundle;
import android.view.Window;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public abstract class BaseActivity extends AppCompatActivity {
    public Context mContext;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        mContext = this;
    }
}

File: kotlin/kotlindemo/src/main/java/com/example/demo/BaseActivity.kt
package com.example.demo

import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlin.properties.Delegates

abstract class BaseActivity : AppCompatActivity() {

    lateinit var mContext: Context
    private val a: String by Delegates.notNull()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(getResId())
        ActivityManager.addActivity(this)
        mContext = this

        initView()
        initData()
        initListener()

    }

    abstract fun getResId(): Int

    abstract fun initView()

    open fun initData() {

    }

    open fun initListener() {

    }

    override fun onDestroy() {
        super.onDestroy()
        ActivityManager.removeActivity(this)
    }
}